Subject: [PATCH] a
---
Index: regex_3/_regex_unicode.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/regex_3/_regex_unicode.h b/regex_3/_regex_unicode.h
--- a/regex_3/_regex_unicode.h	(revision 8ced3047298b1824c1c741183013cd6c1ff3fd07)
+++ b/regex_3/_regex_unicode.h	(date 1743642412321)
@@ -5,7 +5,6 @@
 typedef unsigned int RE_UINT32;
 typedef signed int RE_INT32;
 
-typedef unsigned char BOOL;
 #if !defined(FALSE) || !defined(TRUE)
 #define FALSE 0
 #define TRUE 1
