diff --git a/py/src/constraint.cpp b/py/src/constraint.cpp
index f2ac2a2..94818fc 100644
--- a/py/src/constraint.cpp
+++ b/py/src/constraint.cpp
@@ -5,9 +5,9 @@
 |
 | The full license is in the file LICENSE, distributed with this software.
 |----------------------------------------------------------------------------*/
+#include <cppy/cppy.h>
 #include <algorithm>
 #include <sstream>
-#include <cppy/cppy.h>
 #include <kiwi/kiwi.h>
 #include "types.h"
 #include "util.h"
diff --git a/py/src/expression.cpp b/py/src/expression.cpp
index a53cc1f..ff8bd63 100644
--- a/py/src/expression.cpp
+++ b/py/src/expression.cpp
@@ -5,8 +5,8 @@
 |
 | The full license is in the file LICENSE, distributed with this software.
 |----------------------------------------------------------------------------*/
-#include <sstream>
 #include <cppy/cppy.h>
+#include <sstream>
 #include "symbolics.h"
 #include "types.h"
 #include "util.h"
diff --git a/py/src/kiwisolver.cpp b/py/src/kiwisolver.cpp
index c6f2d55..b7c4233 100644
--- a/py/src/kiwisolver.cpp
+++ b/py/src/kiwisolver.cpp
@@ -5,8 +5,8 @@
 |
 | The full license is in the file LICENSE, distributed with this software.
 |----------------------------------------------------------------------------*/
-#include <mutex>
 #include <cppy/cppy.h>
+#include <mutex>
 #include <kiwi/kiwi.h>
 #include "types.h"
 #include "version.h"
diff --git a/py/src/solver.cpp b/py/src/solver.cpp
index 0dc0fdf..2bc224d 100644
--- a/py/src/solver.cpp
+++ b/py/src/solver.cpp
@@ -198,7 +198,7 @@ PyObject*
 Solver_dump( Solver* self )
 {
 	cppy::ptr dump_str( PyUnicode_FromString( self->solver.dumps().c_str() ) );
-	PyObject_Print( dump_str.get(), stdout, 0 );
+	PyObject_Print( dump_str.get(), (EFILE*)stdout, 0 );
 	Py_RETURN_NONE;
 }

diff --git a/py/src/term.cpp b/py/src/term.cpp
index 69fbab8..cbf12c9 100644
--- a/py/src/term.cpp
+++ b/py/src/term.cpp
@@ -5,8 +5,8 @@
 |
 | The full license is in the file LICENSE, distributed with this software.
 |----------------------------------------------------------------------------*/
-#include <sstream>
 #include <cppy/cppy.h>
+#include <sstream>
 #include "symbolics.h"
 #include "types.h"
 #include "util.h"
diff --git a/py/src/util.h b/py/src/util.h
index ba6a940..0535ab4 100644
--- a/py/src/util.h
+++ b/py/src/util.h
@@ -6,10 +6,10 @@
 | The full license is in the file LICENSE, distributed with this software.
 |----------------------------------------------------------------------------*/
 #pragma once
+#include <cppy/cppy.h>
 #include <map>
 #include <mutex>
 #include <string>
-#include <cppy/cppy.h>
 #include <kiwi/kiwi.h>
 #include "types.h"
 
diff --git a/pyproject.toml b/pyproject.toml
index 3d39348..e4043aa 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -36,7 +36,6 @@
 [build-system]
   requires = [
     "setuptools>=61.2; implementation_name == 'cpython'",
-    "setuptools>=61.2,<=72.1.0; implementation_name != 'cpython'",
     "setuptools_scm[toml]>=3.4.3",
     "cppy>=1.3.0"
   ]
