diff --git a/mesonpy/__init__.py b/mesonpy/__init__.py
index 4d785a3..10f28db 100644
--- a/mesonpy/__init__.py
+++ b/mesonpy/__init__.py
@@ -75,7 +75,7 @@ _MESON_ARGS_KEYS = ['dist', 'setup', 'compile', 'install']

 _SUFFIXES = importlib.machinery.all_suffixes()
 _EXTENSION_SUFFIX_REGEX = re.compile(r'^[^.]+\.(?:(?P<abi>[^.]+)\.)?(?:so|pyd|dll)$')
-assert all(re.match(_EXTENSION_SUFFIX_REGEX, f'foo{x}') for x in importlib.machinery.EXTENSION_SUFFIXES)
+#assert all(re.match(_EXTENSION_SUFFIX_REGEX, f'foo{x}') for x in importlib.machinery.EXTENSION_SUFFIXES)

 # Map Meson installation path placeholders to wheel installation paths.
 # See https://docs.python.org/3/library/sysconfig.html#installation-paths
@@ -345,7 +345,8 @@ class _WheelBuilder():
             # distribute any file in {platlib}) thus use generic
             # implementation and ABI tags.
             return mesonpy._tags.Tag('py3', 'none', None)
-        return mesonpy._tags.Tag(None, self._stable_abi, None)
+        import packaging
+        return list(packaging.tags.generic_tags())[0]

     @property
     def name(self) -> str:
@@ -409,7 +410,7 @@ class _WheelBuilder():
         # ABI, therefore extension modules using the limited API do
         # not use the stable ABI filename suffix and wheels should not
         # be tagged with the abi3 tag.
-        if self._limited_api and '__pypy__' not in sys.builtin_module_names:
+        if self._limited_api and '__pypy__' not in sys.builtin_module_names and sys.implementation.name != 'nuitkapython':
             # Verify stable ABI compatibility: examine files installed
             # in {platlib} that look like extension modules, and raise
             # an exception if any of them has a Python version
@@ -785,7 +786,8 @@ class Project():
             # default build options
             '-Dbuildtype=release',
             '-Db_ndebug=if-release',
-            '-Db_vscrt=md',
+            '-Db_vscrt=mt',
+            '-Dprefer_static=True',
             # user build options
             *self._meson_args['setup'],
             # pass native file last to have it override the python
