diff --git a/meson.build b/meson.build
index d816cca..2878318 100644
--- a/meson.build
+++ b/meson.build
@@ -13,6 +13,8 @@ project(
     'c_std=c11',
     'cpp_std=c++17',
     'pkgconfig.relocatable=true',
+    'prefer_static=True',
+    'b_vscrt=mt',
   ],
 )

diff --git a/numpy/_core/meson.build b/numpy/_core/meson.build
index d32d71a..2f22343 100644
--- a/numpy/_core/meson.build
+++ b/numpy/_core/meson.build
@@ -740,29 +740,6 @@ _umath_tests_mtargets = mod_features.multi_targets(
   dependencies: [py_dep, np_core_dep]
 )

-test_modules_src = [
-  ['_umath_tests', [
-      src_file.process('src/umath/_umath_tests.c.src'),
-      'src/common/npy_cpu_features.c',
-    ],
-    _umath_tests_mtargets.static_lib('_umath_tests_mtargets')
-  ],
-  ['_rational_tests', 'src/umath/_rational_tests.c', []],
-  ['_struct_ufunc_tests', 'src/umath/_struct_ufunc_tests.c', []],
-  ['_operand_flag_tests', 'src/umath/_operand_flag_tests.c', []],
-]
-foreach gen: test_modules_src
-  py.extension_module(gen[0],
-    gen[1],
-    c_args: c_args_common,
-    include_directories: ['src/multiarray', 'src/npymath'],
-    dependencies: np_core_dep,
-    install: true,
-    subdir: 'numpy/_core',
-    link_with: gen[2],
-    install_tag: 'tests'
-  )
-endforeach

 # Build multiarray dispatch-able sources
 # --------------------------------------
diff --git a/numpy/_core/src/common/numpyos.c b/numpy/_core/src/common/numpyos.c
index a5ca280..e33e5d1 100644
--- a/numpy/_core/src/common/numpyos.c
+++ b/numpy/_core/src/common/numpyos.c
@@ -814,14 +814,7 @@ NumPyOS_strtoull(const char *str, char **endptr, int base)
  * earlier than 14.0 (_MSC_VER == 1900).
  */

-static void __cdecl _silent_invalid_parameter_handler(
-    wchar_t const* expression,
-    wchar_t const* function,
-    wchar_t const* file,
-    unsigned int line,
-    uintptr_t pReserved) { }
-
-_invalid_parameter_handler _Py_silent_invalid_parameter_handler = _silent_invalid_parameter_handler;
+

 #endif

diff --git a/numpy/distutils/command/build_ext.py b/numpy/distutils/command/build_ext.py
index 5c62d90..4e243f9 100644
--- a/numpy/distutils/command/build_ext.py
+++ b/numpy/distutils/command/build_ext.py
@@ -617,15 +617,36 @@ def build_extension(self, ext):
                     fcompiler, library_dirs,
                     unlinkable_fobjects)

-        linker(objects, ext_filename,
-               libraries=libraries,
-               library_dirs=library_dirs,
-               runtime_library_dirs=ext.runtime_library_dirs,
-               extra_postargs=extra_args,
-               export_symbols=self.get_export_symbols(ext),
-               debug=self.debug,
-               build_temp=self.build_temp,
-               target_lang=ext.language)
+        ext_filename = ext_filename.replace(".so", "")
+
+        self.compiler.create_static_lib(
+            objects, ext_filename,
+            output_dir=os.path.abspath("."),
+            debug=self.debug,
+            target_lang=ext.language)
+
+        result_path = \
+            self.compiler.library_filename(ext_filename,
+                                           output_dir=os.path.abspath("."))
+
+        with open(result_path + '.link.json', 'w') as f:
+            json.dump({
+                'libraries': self.get_libraries(ext),
+                'library_dirs': ext.library_dirs + ['lib'],
+                'runtime_library_dirs': ext.runtime_library_dirs,
+                'extra_postargs': extra_args}, f)
+
+        for lib in self.get_libraries(ext):
+            for dir in ext.library_dirs:
+                lib_install_dir = os.path.join(os.path.dirname(ext_path), dir)
+                print(os.path.join(ext_path, dir, lib + '.lib'))
+                if os.path.isfile(os.path.join(dir, lib + '.lib')):
+                    if not os.path.isabs(dir):
+                        if not os.path.exists(lib_install_dir):
+                            os.makedirs(lib_install_dir)
+                        shutil.copyfile(os.path.join(dir, lib + '.lib'),
+                                        os.path.join(lib_install_dir, lib + '.lib'))
+                    break

     def _add_dummy_mingwex_sym(self, c_sources):
         build_src = self.get_finalized_command("build_src").build_src
diff --git a/vendored-meson/meson/mesonbuild/build.py b/vendored-meson/meson/mesonbuild/build.py
index 02d2bb9..cc3fc56 100644
--- a/vendored-meson/meson/mesonbuild/build.py
+++ b/vendored-meson/meson/mesonbuild/build.py
@@ -1448,12 +1448,11 @@ def get_internal_static_libraries(self) -> OrderedSet[BuildTargetTypes]:
     def get_internal_static_libraries_recurse(self, result: OrderedSet[BuildTargetTypes]) -> None:
         for t in self.link_targets:
             if t.is_internal() and t not in result:
-                self.check_can_extract_objects(t, origin, promoted=True)
                 result.add(t)
-                t.get_internal_static_libraries_recurse(result, origin)
+                t.get_internal_static_libraries_recurse(result)
         for t in self.link_whole_targets:
             if t.is_internal():
-                t.get_internal_static_libraries_recurse(result, origin)
+                t.get_internal_static_libraries_recurse(result)

     def check_can_extract_objects(self, t: T.Union[Target, CustomTargetIndex], origin: StaticLibrary, promoted: bool = False) -> None:
         if isinstance(t, (CustomTarget, CustomTargetIndex)) or t.uses_rust():
diff --git a/vendored-meson/meson/mesonbuild/dependencies/python.py b/vendored-meson/meson/mesonbuild/dependencies/python.py
index 46d12f3..5e377cd 100644
--- a/vendored-meson/meson/mesonbuild/dependencies/python.py
+++ b/vendored-meson/meson/mesonbuild/dependencies/python.py
@@ -222,7 +222,7 @@ def get_windows_link_args(self, limited_api: bool) -> T.Optional[T.List[str]]:
             verdot = self.variables.get('py_version_short')
             imp_lower = self.variables.get('implementation_lower', 'python')
             if self.static:
-                libpath = Path('libs') / f'libpython{vernum}.a'
+                libpath = Path('libs') / f'python{vernum}.lib'
             else:
                 if limited_api:
                     vernum = vernum[0]
diff --git a/vendored-meson/meson/mesonbuild/modules/python.py b/vendored-meson/meson/mesonbuild/modules/python.py
index 30fc50b..7d2189b 100644
--- a/vendored-meson/meson/mesonbuild/modules/python.py
+++ b/vendored-meson/meson/mesonbuild/modules/python.py
@@ -10,7 +10,7 @@
 from .. import mesonlib
 from .. import mlog
 from ..options import UserFeatureOption
-from ..build import known_shmod_kwargs, CustomTarget, CustomTargetIndex, BuildTarget, GeneratedList, StructuredSources, ExtractedObjects, SharedModule
+from ..build import known_shmod_kwargs, CustomTarget, CustomTargetIndex, BuildTarget, GeneratedList, StructuredSources, ExtractedObjects, SharedModule, StaticLibrary
 from ..dependencies import NotFoundDependency
 from ..dependencies.detect import get_dep_identifier, find_external_dependency
 from ..dependencies.python import BasicPythonExternalProgram, python_factory, _PythonDependencyBase
@@ -229,7 +229,7 @@ def extension_module_method(self, args: T.Tuple[str, T.List[BuildTargetSource]],
                 (self.is_pypy or mesonlib.version_compare(self.version, '>=3.9')):
             kwargs['gnu_symbol_visibility'] = 'inlineshidden'

-        return self.interpreter.build_target(self.current_node, args, kwargs, SharedModule)
+        return self.interpreter.build_target(self.current_node, args, kwargs, StaticLibrary)

     def _convert_api_version_to_py_version_hex(self, api_version: str, detected_version: str) -> str:
         python_api_version_format = re.compile(r'[0-9]\.[0-9]{1,2}')
diff --git a/vendored-meson/meson/mesonbuild/modules/python3.py b/vendored-meson/meson/mesonbuild/modules/python3.py
index 45ad850..11ebcf0 100644
--- a/vendored-meson/meson/mesonbuild/modules/python3.py
+++ b/vendored-meson/meson/mesonbuild/modules/python3.py
@@ -10,7 +10,7 @@
 from . import ExtensionModule, ModuleInfo, ModuleState
 from ..build import (
     BuildTarget, CustomTarget, CustomTargetIndex, ExtractedObjects,
-    GeneratedList, SharedModule, StructuredSources, known_shmod_kwargs
+    GeneratedList, SharedModule, StaticLibrary, StructuredSources, known_shmod_kwargs
 )
 from ..interpreter.type_checking import SHARED_MOD_KWS
 from ..interpreterbase import typed_kwargs, typed_pos_args, noPosargs, noKwargs, permittedKwargs
@@ -52,7 +52,7 @@ def extension_module(self, state: ModuleState, args: T.Tuple[str, T.List[BuildTa
             suffix = []
         kwargs['name_prefix'] = ''
         kwargs['name_suffix'] = suffix
-        return self.interpreter.build_target(state.current_node, args, kwargs, SharedModule)
+        return self.interpreter.build_target(state.current_node, args, kwargs, StaticLibrary)

     @noPosargs
     @noKwargs
diff --git a/numpy/linalg/meson.build b/numpy/linalg/meson.build
index 740c9f5..6341fdb 100644
--- a/numpy/linalg/meson.build
+++ b/numpy/linalg/meson.build
@@ -3,18 +3,7 @@
 python_xerbla_sources = ['lapack_lite/python_xerbla.c']

 lapack_lite_sources = []
-if not have_lapack
-  lapack_lite_sources = [
-    'lapack_lite/f2c.c',
-    'lapack_lite/f2c_c_lapack.c',
-    'lapack_lite/f2c_d_lapack.c',
-    'lapack_lite/f2c_s_lapack.c',
-    'lapack_lite/f2c_z_lapack.c',
-    'lapack_lite/f2c_blas.c',
-    'lapack_lite/f2c_config.c',
-    'lapack_lite/f2c_lapack.c',
-  ]
-endif
+

 py.extension_module('lapack_lite',
   [
diff --git a/numpy/f2py/src/fortranobject.h b/numpy/f2py/src/fortranobject.h
index 4aed2f6..53d4012 100644
--- a/numpy/f2py/src/fortranobject.h
+++ b/numpy/f2py/src/fortranobject.h
@@ -12,7 +12,9 @@ extern "C" {
 #ifdef FORTRANOBJECT_C
 #define NO_IMPORT_ARRAY
 #endif
+#ifndef PY_ARRAY_UNIQUE_SYMBOL
 #define PY_ARRAY_UNIQUE_SYMBOL _npy_f2py_ARRAY_API
+#endif
 #include "numpy/arrayobject.h"
 #include "numpy/npy_3kcompat.h"

